torch/utils/data/dataloader.pyi

```diff
@@ -28,12 +28,10 @@ class DataLoader(Generic[T_co]):
                  worker_init_fn: _worker_init_fn_t=...) -> None: ...
 
     def __len__(self) -> int: ...
-    # We quote '_DataLoaderIter' since it isn't defined yet and the definition can't be moved up since
-    # '_DataLoaderIter' references 'DataLoader'. Pending updates of PEP 484 will fix this.
-    def __iter__(self) -> '_DataLoaderIter':...
+    def __iter__(self) -> _BaseDataLoaderIter:...
```

Comment:
Has the problem mentioned in previous comment been solved?

Reply:
I think so? I wasn’t able to reproduce the error in the comments with the latest version of mypy as well as a version from April, I was running:

```
mypy torch/utils/data/dataloader.py
```

and didn’t see any errors from mypy. That invocation may not have been sufficient to trigger the error @malmaud saw when they originally wrote this code.

Reply:
At some point relatively recently, Python/mypy added the ability to handle unquoted forward references, but there are definitely versions which cannot. I think it's safer to keep the quotes unless I was actually in error before and this isn't a forward reference.

Reply:
@ngoldbaum mind updating this? :)

Reply:
Done, I also expanded the comment to reflect the current situation in mypy so future readers are hopefully a little less confused.